src/extension/iSamples_resultList.js

```diff
@@ -31,10 +34,42 @@ class ResultList extends React.Component {
     this.setState({hiddenCols : updatedHiddenCols}) 
   }
 
+  // compare two arrays and checks equality 
+  arraysEqual(a, b) {
+    if (a === b) return true;
+    if (a == null || b == null) return false;
+    if (a.length !== b.length) return false;
+  
+    for (var i = 0; i < a.length; ++i) {
+      if (a[i] !== b[i]) return false;
+    }
+    return true;
+  }
+
   // prevent unnecessary rerendering
   shouldComponentUpdate(nextProps) {
     this.switchView(store.getState()['query']['view']['facet']);
-    if (JSON.stringify(nextProps) !== JSON.stringify(this.props)) {
+
```

Comment:
After some debugging, I found that this was returning true whenever we clicked the facet, which caused the entire component to rerender.